design/data-distributor-internals.md

```diff
@@ -6,78 +6,102 @@ Data distribution manages the lifetime of storage servers, decides which storage
+When a data distribution role is created, it recovers the states of the previous DD from the system keyspace. First, it sets itself as the owner of the `moveKeysLock`. Then it collects the information of servers and shards, the map between servers and shards, and the replication configuration by reading DD-related system keyspace (i.e., *serverKeys* sub-space). Based on the information, the new DD recreates its components (e.g., servers, teams, and trackers) that matches the states of the previous DD. Trackers will evaluate the healthiness of servers and teams based on the replication policy. Unhealthy servers and teams will be removed and new servers and teams will be created if the replication policy is changed.
 
 ### When to move keys?
 
-Keys can be moved from a server to another for several reasons: (a) DD moves keys from overutilized servers to underutilized servers, where a server’s utilization is defined as the server’s disk usage; (b) DD splits or merges shards in order to rebalance the disk usage of servers; (c) DD removes redundant teams when the team number is larger than the desired number; (d) DD repairs the replication factor by duplicate shards from a server to another when servers in a team fail.
+Keys can be moved from a server to another for several reasons:
 
-Actors are created to monitor the reasons of key movement: (a) MountainChopper and ValleyFiller actors periodically measure a random server team’s utilization and rebalance the server’s keys among other servers; (b) shardMerger and shardSplitter actors take a shard as input and respectively evaluates if the input shard can be merged with its neighboring shards without creating a too big shard and if the shard should be split. Once new shards are created, the actors create the shard’s tracker and send RelocateShard requests to DD’s queue; (c) serverTeamRemover and machineTeamRemover actors periodically evaluate if the number of server teams and machine teams is larger than the desired number. If so, they respectively pick a server team or a machine team to remove based on predefined criteria; (d) teamTracker actor monitors a team’s healthiness. When a server in the team becomes unhealthy, it issues the RelocateShard request to repair the replication factor. The less servers a team has, the higher priority the RelocateShard request will be.
+* DD moves keys from overutilized servers to underutilized servers, where a server’s utilization is defined as the server’s disk usage;
```

Comment:
I personally don't like listing it in bullet items, which makes the items look unnecessarily more significant that other parts.  
I commented on that in the review process and did this on purpose.

Another way to think this is: if we write this in a paper or technical report, will you put all such items in a bullet list? From my experience, I won't unless the item is something I want to highlight.

Reply:
I made into bullets because staring at a large paragraph on screen is hurting my eyes. :) I'll revert back to paragraph and make the items using a consistent style.